### PR TITLE
Initial implementation of virtfs-ext4

### PR DESCRIPTION
**Pull request by movitto**
Refactors & updates existing logic to conform to standards and implement virtfs interface. specs are also included

### CODE REVIEW

**Comment by movitto** [comment]
@roliveri this is ready for review

**Comment by roliveri** [comment]
@hsong-rh @jerryk55 Please join in on the review.

File: lib/virtfs/ext4/extent.rb [hunk 1]
```diff
@@ -23,13 +21,5 @@ def initialize(buf)
       @length     = @extent['length']
       @start      = (@extent['start_hi'] << 32) | @extent['start_lo']
     end
-
```
Comment (jerryk55) [inline lib/virtfs/ext4/extent.rb:26]: I'll make the comment only once in this PR - I would prefer it if you left the dump methods intact.

File: lib/virtfs/ext4/superblock.rb [hunk 2]
```diff
@@ -142,271 +138,171 @@ class Superblock
...
 
       # Seek, read & decode the superblock structure
       @stream.seek(SUPERBLOCK_OFFSET)
       @sb = SUPERBLOCK.decode(@stream.read(SUPERBLOCK_SIZE))
 
       # Grab some quick facts & make sure there's nothing wrong. Tight qualification.
-      raise "Ext4::Superblock.initialize: Invalid signature=[#{@sb['signature']}]" if @sb['signature'] != SUPERBLOCK_SIG
-      state = @sb['fs_state']
-      raise "Ext4::Superblock.initialize: Invalid file system state" if state > FSS_END
-      if state != FSS_CLEAN
```
Comment (jerryk55) [inline lib/virtfs/ext4/superblock.rb:162]: Why was this error logging removed?

**Comment by jerryk55** [comment]
A comment on this PR and the Ext3 PR - it seems that you've changed many of the exceptions raised to remove the specific module::class.method name from the messages.  But not all.  First question is why were you changing these, and the second question is why aren't you being consistent about it.  Thanks..

**Review by jerryk55 — COMMENTED** [review]
It seems as if you are removing all error logging.  Is this wise?

File: lib/virtfs/ext4/base.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+module VirtFS::Ext4
+  class FS < VirtFS::Ext3::FS
+
+    attr_accessor :mount_point, :superblock, :root_dir
```
Comment (jerryk55) [inline lib/virtfs/ext4/base.rb:4]: Where is mount_point initialized?

**Pull request closed without merging**